cfme/physical/physical_switch.py

```diff
@@ -0,0 +1,231 @@
+    def serial_number(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Serial Number")
+
+    @variable(alias='ui')
+    def description(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Description")
+
+    @variable(alias='ui')
+    def manufacturer(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Manufacturer")
+
+    def _wait_for_state_change(self, desired_state, target, provider, view, timeout=300, delay=10):
+        """Wait for the Physical Switch to reach the desired state. This function waits an amount
+           of time due to wait_for.
+
+        Args:
+            desired_state (str): 'on' or 'off'
+            target (str): The name of the method that most be used to compare with the desired_state
+            provider (object): 'LenovoProvider'
+            view (object): The view that most be refreshed to verify if the value was changed
+            timeout (int): Specify amount of time (in seconds) to wait until TimedOutError is raised
+            delay (int): Specify amount of time (in seconds) to repeat each time.
+        """
+
+        def _is_state_changed():
+            self.refresh(provider, handle_alert=True)
+            return desired_state == getattr(self, target)()
+
+        wait_for(_is_state_changed, fail_func=view.browser.refresh, num_sec=timeout, delay=delay)
+
+    @property
+    def exists(self):
+        """Checks if the physical_switch exists in the UI.
+
+        Returns: :py:class:`bool`
+        """
+        view = navigate_to(self.parent, "All")
+        try:
+            view.entities.get_entity(name=self.name, surf_pages=True)
+        except ItemNotFound:
+            return False
+        else:
+            return True
+
+    @cached_property
+    def db_id(self):
+        return self.appliance.physical_switch_id(self.name)
+
+    def wait_to_appear(self):
+        """Waits for the Switch to appear in the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the Switch to appear...")
+        wait_for(
+            lambda: self.exists,
+            message="Wait for the Switch to appear",
+            num_sec=1000,
+            fail_func=view.browser.refresh
+        )
+
+    def wait_for_delete(self):
+        """Waits for the Switch to remove from the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the Switch to delete...")
+        wait_for(
+            lambda: not self.exists,
+            message="Wait for the Switch to disappear",
+            num_sec=500,
+            fail_func=view.browser.refresh
+        )
+
+
+@attr.s
+class PhysicalSwitchCollection(BaseCollection):
+    """Collection object for the :py:class:`cfme.infrastructure.host.PhysicalSwitch`."""
+
+    ENTITY = PhysicalSwitch
+
+    def select_entity_rows(self, *physical_switches):
```

Comment:
**QUESTION:** I'm guessing these can't be accessed by REST, that would obviously be the better way to "find" them.

Reply:
@psav I can't see how the UI checkboxes could be accessed by REST, is that what you meant?

Reply:
I was more meaning the ```.all()``` function below - can be a new PR

Reply:
@psav access Physical Switches by REST depends on https://github.com/ManageIQ/wrapanapi/pull/294.